Source/NSDrawer.m

```diff
@@ -590,6 +598,46 @@ - (void) dealloc
 }
 
 // Opening and Closing
+- (NSRectEdge) _computeEdge
+{
+  NSRectEdge result = _preferredEdge;
+  NSRect windowFrame = [[self parentWindow] frame];
+  NSRect screenRect = [[NSScreen mainScreen] frame];
+  
+  switch (_preferredEdge)
+    {
+    case NSMinXEdge:
+      if (windowFrame.origin.x < _maxContentSize.width)
+	{
+	  result = NSMaxXEdge;
```

Comment:
Shouldn't we check, whether the space on the right is actually bigger then the space on the left?

Reply:
Well, this is just determining if there is enough space on the Min (left) edge between the window and the edge of the screen.   I wanted to keep the calculation simple.

Reply:
Let's assume that you have a reason why you check that. Wouldn't the same reason apply for the opposing side? If both sides are to small for the drawer, shouldn't we use the bigger one?

Reply:
My thought process at the time was that this scenario (in which we would have to check which side had more space) was an edge case.  I had assumed (probably badly) that if there wasn't space on one side unless we either have a very large window or a really small screen, we could safely open the drawer on the other side.  I do, however, see your point, and the code to cover that case (whether it is an edge case or not) is trivial.   I will go ahead and do it and resolve this once done.

Reply:
Done.

Reply:
Sorry, this is not the condition I did intend. I would expect that we respect that preferred edge, if the drawer fits there, but when it doesn't we use the side where there is the most space.

Reply:
Fred, this does preserve that.  If there is space on the preferredEdge then it opens on that edge.   The condition you're describing IS what is implemented AFAIK.

My test app is here, by the way...[ ](https://github.com/gcasa/NSDrawer_test)

Reply:
That is strange. The code I see her does not look at the size of the drawer at all:

` case NSMinXEdge:
       if (right > left)
 	{
 	  result = NSMaxXEdge;
        }`

Is GitHub showing me the wrong code (this has happened before)?

Reply:
No, that's the right code.  I will re-write it.   I think I see the issue you and, perhaps, Riccardo are talking about.

Reply:
I believe I have the solution now.   Please take a look and test and let me know.  I will solve all other issues (i.e., ones NOT pertaining to overriding the preferredEdge as this change does) in other PRs.